src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -430,28 +430,33 @@ private void RecordRequestStartMetrics(HttpContext httpContext)
             }
         }
 
+        // The trace id was successfully extracted, so we can set the trace state
+        // https://www.w3.org/TR/trace-context/#tracestate-header
         if (!string.IsNullOrEmpty(requestId))
```

Comment:
I suspect that checking `requestId` has probably become redundant, but I'm fine keeping it in the spirit of making a conservative change.

Reply:
Agreed that it is probably redundant, and also agreed that keeping the check anyways is the more conservative change :)